nifi-mock/src/main/java/org/apache/nifi/util/MockProcessSession.java

```diff
@@ -458,6 +458,9 @@ public MockFlowFile importFrom(final Path path, final boolean keepSourceFile, Fl
 
         newFlowFile.setData(baos.toByteArray());
         newFlowFile = putAttribute(newFlowFile, CoreAttributes.FILENAME.key(), path.getFileName().toString());
+        if (!keepSourceFile) {
```

Comment:
Was this included on accident? Does not seem relevant to your ticket.

Reply:
It is necessary, in that the mock was ignoring the attribute, so when tested with multiple files, behaviour was non-deterministic, as it could quite easily pick up the same file again.